app/models/group.rb

```diff
@@ -0,0 +1,4 @@
+class Group < ApplicationRecord
+  belongs_to :user
+  has_and_belongs_to_many :expenditures, dependent: :destroy
+end
```

Comment:
- [ ] Refer to the below association to get an idea of how you can implement your associations with the Group model.

 `` has_many :expenditures, foreign_key: 'group_id', class_name: 'Expenditures', dependent: :destroy``
     ``belongs_to: user``
     
    -  Please add validation to your project for example.
    
  ``validates_format_of :icon, with: /\.(gif|jpe?g|png)$/i,
                             message: 'must have an image extension (jpg, jpe, png, gif)', multiline: true``
  `` validates :name, length: { in: 3..20 } ``
  `` validates :icon, presence: true ``